porting-a-new-endpoint.md

```diff
@@ -0,0 +1,201 @@
+
+### New Service Method
+
+In [pkg/contract/service/tracking.g.go](pkg/contract/service/tracking.g.go), the interface has a new function that we need to implement, which is used in the routes file above.
+
+```go
+type TrackingService interface {
+    ...
+	DeleteTag(ctx context.Context, input *protos.DeleteTag) (*protos.DeleteTag_Response, *contract.Error)
+}
+```
+
+### New FFI Endpoint
+
+In [pkg/lib/tracking.g.go](pkg/lib/tracking.g.go), we need to expose that same service method so it can be called directly from Python via FFI. More on that later.
+
+```go
+//export TrackingServiceDeleteTag
+func TrackingServiceDeleteTag(serviceID int64, requestData unsafe.Pointer, requestSize C.int, responseSize *C.int) unsafe.Pointer {
+	service, err := trackingServices.Get(serviceID)
+	if err != nil {
+		return makePointerFromError(err, responseSize)
+	}
+	return invokeServiceMethod(service.DeleteTag, new(protos.DeleteTag), requestData, requestSize, responseSize)
+}
+```
+
+## Validate Input
+
+A first step to port an endpoint would be to check the request validation happening at the HTTP level. Open [handlers.py](.mlflow.repo/mlflow/server/handlers.py) (from the `.mlflow.repo`) to see which fields are required.
+
+```python
+@catch_mlflow_exception
+@_disable_if_artifacts_only
+def _delete_tag():
+    request_message = _get_request_message(
+        DeleteTag(),
+        schema={
+            "run_id": [_assert_required, _assert_string],
+            "key": [_assert_required, _assert_string],
+        },
+    )
+    _get_tracking_store().delete_tag(request_message.run_id, request_message.key)
+    response_message = DeleteTag.Response()
+    response = Response(mimetype="application/json")
+    response.set_data(message_to_json(response_message))
+    return response
+```
+
+We notice that `run_id` and `key` are required. To ensure parity, we need to update the input struct `DeleteTag` from the generated proto code ([pkg/protos/service.pb.go](pkg/protos/service.pb.go)). ⚠️ Since this file was generated, we don't want to modify it directly. Instead, we will configure validations in [validations.go](magefiles/generate/validations.go).
+
+```go
+var validations = map[string]string{
+    ...
+	"DeleteTag_RunId":                    "required",
+	"DeleteTag_Key":                      "required",
+}
+```
+
+After that, run `mage generate` again and check if our fields in `DeleteTag` now contain `validate:"required"`.
+
+## Implement the Service Method
+
+We aim to keep the Go implementation as close to the Python code as possible. That's why we have a thin service layer and keep most logic in the (SQL) store instead of the service. This keeps things similar to Python, making it easier to port and compare the code.
+
+In the service, we need to call the store with the same arguments as the Python store would be called. This is typically where we convert proto structs to entities. In the case of `deleteTag`, we can just pass the two arguments to the store.
+
```

Comment:
It would be good to explain which file to edit, and whether we need to create a new file or not (in this case, we had to create the `/pkg/tracking/service/tags.go` file since there weren't any other tag endpoints created yet).